Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -1522,6 +1477,64 @@ namespace OpenInfraPlatform {
 				return paths;
 			}
 
+			// Function 10: Get rectangle coordiantes
+			
+			void GetRectangleCoordinates (std::vector<carve::geom::vector<2>>& coords, double radius, double x, double y) const
+			{
+
+				if (radius != 0) {
+					addArc(coords, radius, 0, M_PI_2, x * 0.5 - radius, y * 0.5 - radius);
+					addArc(coords, radius, M_PI_2, M_PI_2, -x * 0.5 + radius, y * 0.5 - radius);
+					addArc(coords, radius, M_PI, M_PI_2, -x * 0.5 + radius, -y * 0.5 + radius);
+					addArc(coords, radius, 3 * M_PI_2, M_PI_2, x * 0.5 - radius, -y * 0.5 + radius);
+				}
+				else {
+					coords.push_back(carve::geom::VECTOR(-x * 0.5, -y * 0.5));
+					coords.push_back(carve::geom::VECTOR(x * 0.5, -y * 0.5));
+					coords.push_back(carve::geom::VECTOR(x * 0.5, y * 0.5));
+					coords.push_back(carve::geom::VECTOR(-x * 0.5, y * 0.5));
+				}
+				
+			}
+
+
+		
+
+			//Function 11: GetTrapeziumCoordinates
+			void GetTrapeziumCoordinates(std::vector<carve::geom::vector<2>>& coords, double xBottom, double xOffset, double xTop, double y)const
+			{
+				coords.push_back(carve::geom::VECTOR(-xBottom * 0.5, -y * 0.5));
+				coords.push_back(carve::geom::VECTOR(xBottom * 0.5, -y * 0.5));
+				coords.push_back(carve::geom::VECTOR(-xBottom * 0.5 + xOffset + xTop, y * 0.5));
+				coords.push_back(carve::geom::VECTOR(-xBottom * 0.5 + xOffset, y * 0.5));
+			}
+
+			//Function 12: GetCircleRadius
+			void GetCircleCoordinates(std::vector<carve::geom::vector<2>>& coords, double radius, double d_angle, int num_segments ) const
+			{
+				
+				double angle = 0;
+				for (int i = 0; i < num_segments; ++i) {
+					coords.push_back(carve::geom::VECTOR((radius * cos(angle)), (radius * sin(angle))));
+					angle += d_angle;
+				}
+				

```

Comment:
For example, also this line break is unnecessary.

Reply:
Done